> * tests\io_xml_test.py asserts `xml_ts.time_series[0].header.start_date == xml_ts.time_series[0].events.index[0]`
>   `Headers.start_date` is of type `datetime.datetime` while `events.index` is of type `pd.Timestamp`. @DanielTollenaar can you make a decision on this?

Don't think this is a decision to make as pd.Timestamp is a subclass of datetime.datetime? The problem is:

```
 AssertionError: assert datetime.datetime(2025, 4, 1, 0, 0) == Timestamp('2025-04-15 03:15:00')
```

The AssertionError is correct. `assert datetime.datetime(2025, 4, 1, 0, 0) == Timestamp('2025-04-01 00:00:00')` does not raise an Error. 

Don't know why this test didn't fail previously, but in hindsight; startdate should be <= first event timestamp and enddate should be >= last event timestamp.

--- a/fewspy/io/read_netcdf.py
+++ b/fewspy/io/read_netcdf.py
@@ -6,10 +6,7 @@ import zipfile
 from io import BytesIO
 import tempfile
 import os
-
-
-type = "instantaneous"
-module_instance_id = "Vullingsgraad"
+import warnings
 
 
 def _parse_time(time_var):
@@ -84,13 +81,22 @@ def read_netcdf(
 
     Args:
         nc_file (Path): path to the NetCDF file
-        time_series_type (str | None, optional): type for timeseries header. Defaults to None. 
-        # TODO wvg; In fewspy\time_series.py Header this is a Literal["accumulative", "instantaneous"]. This should be consistent here.
+        time_series_type (str | None, optional): type for timeseries header. Defaults to None.
+        Note (!) specifying time_series_type is advised. If you don't data will be interpreted as instantaneous
         module_instance_id (str | None, optional): ModuleInstanceId for timeseries header. Defaults to None.
 
     Returns:
         TimeSeriesSet: timeseries
     """
+    if time_series_type is None:
+        warnings.warn(
+            "time_series_type is None; defaulting to 'instantaneous'. "
+            "Pass time_series_type explicitly to avoid this warning.",
+            UserWarning,
+            stacklevel=2,
+        )
+        time_series_type = "instantaneous"
+
     # Read file
     with Dataset(nc_file, mode="r") as ds:
         # init TimeSeriesSet

--- a/fewspy/io/read_xml.py
+++ b/fewspy/io/read_xml.py
@@ -53,13 +53,6 @@ def _parse_xml_data(root) -> TimeSeriesSet:
             else:
                 data += [{k: v for k, v in zip(subchild.keys(), subchild.values())}]
 
-        # add to time_series_set
-        # TODO wvg; dit kan weg, startDate en endDate zijn verplichte velden in Header.
-        # if "startDate" not in metadata:
-        #     metadata["startDate"] = {"date": data[0]["date"], "time": data[0]["time"]}
-        # if "endDate" not in metadata:
-        #     metadata["endDate"] = {"date": data[-1]["date"], "time": data[-1]["time"]}
-
         time_series_set["timeSeries"] += [{"header": metadata, "events": data}]
 
     return TimeSeriesSet.from_dict(time_series_set)

--- a/tests/io_xml_test.py
+++ b/tests/io_xml_test.py
@@ -1,4 +1,5 @@
 # %%
+from time import time
 import fewspy
 from config import data_dir
 
@@ -32,9 +33,9 @@ def test_xml_ts():
 
     # xcheck start_time and end_time
     expected_start_date = xml_ts.time_series[0].events.index[0]
-    assert xml_ts.time_series[0].header.start_date == expected_start_date
+    assert xml_ts.time_series[0].header.start_date <= expected_start_date
     expected_end_date = xml_ts.time_series[0].events.index[-1]
-    assert xml_ts.time_series[0].header.end_date == expected_end_date
+    assert xml_ts.time_series[0].header.end_date >= expected_end_date
 
     # xcheck interval
     expected_time_step = (
@@ -109,12 +110,13 @@ def test_netcdf_ts():
 
     Therefore we don't compare these info
     """
-    nc_ts = fewspy.read_netcdf(NC_FILE)
-
+    nc_ts = fewspy.read_netcdf(
+        NC_FILE, time_series_type="instantaneous", module_instance_id="Vullingsgraad"
+    )
 
     assert nc_ts.time_zone == 0.0
     # headers
-    ignore_keys = ["type", "module_instance_id", "z"]
+    ignore_keys = ["type", "module_instance_id", "z", "start_date", "end_date"]
 
     nc_header = {
         k: v

